ecommerce/api.py

```diff
@@ -530,7 +531,10 @@ def enroll_user_in_order_items(order):
             voucher.save()
     for program in programs:
         enrollment, created = ProgramEnrollment.all_objects.get_or_create(
-            user=order.purchaser, program=program, defaults=dict(company=company)
+            user=order.purchaser,
```

Comment:
Question: it looks like we're not treating `(user|order|program)` as a unique set of fields for the enrollment, but there's not actually a db constraint on these so it's possible for this to raise an error. So I think we should probably be defining a `unique_together` to ensure we don't get a `MultipleObjectsReturned` out of this. I'm not sure whether that's just `user|program` as it was originally or if `order` should now be part of that. Guess that ultimately comes down to whether multiple program enrollments are/should be possible.

I think all of the above also holds true for `CourseRunEnrollment`.

Reply:
✔️